src/core/services/theming/theming.js

```diff
@@ -597,6 +679,50 @@ function parseRules(theme, colorType, rules) {
     generatedRules.push(newRule);
   });
 
+  var inverseTypes = ['primary', 'accent', 'warn']
+  angular.forEach(inverseTypes, function(colorType) {
+    rules = originalRules;
+    rules = rules.replace(/THEME_NAME/g, theme.name);
+    angular.forEach(color.hues, function(hueValue, hueName) {
+      var parseCSSRulesRegex = /(?:\/\*.+?\*\/)?(.+?)\{(.+?)\}(?!(\}|'|"|;))/g;
+      var regexMatch;
+      var relatedRules = '';
+      while ((regexMatch = parseCSSRulesRegex.exec(rules)) !== null) {
+        if (regexMatch.index === parseCSSRulesRegex.lastIndex) {
+          regexMatch.lastIndex++;
+        }
+        var selectors = [];
+        var colorTypeClass = '.md-' + colorType;
+        var hueNameClass = hueName === 'default' ? '' : '.md-' + hueName;
+        angular.forEach(regexMatch[1].split(/,/g), function(selector) {
```

Comment:
forEach within while within forEach within forEach

This must be suuuper expensive, we should come up with better idea how to do it